### PR TITLE
Integrate try-runtime into substrate node template

### LINKED ISSUE #10892: Integrate `try-runtime` into `substrate-node-template`
Add `TryRuntime` subcommand here: 

https://github.com/paritytech/substrate/blob/917fe4919c70cb022a9c90eb6415cf85cbc67f7b/bin/node-template/node/src/cli.rs#L13

in a similar fashion to: 

https://github.com/paritytech/substrate/blob/917fe4919c70cb022a9c90eb6415cf85cbc67f7b/bin/node/cli/src/cli.rs#L46

### PR DESCRIPTION
**Pull request by omadoyeabraham**
Integrated try-runtime into the node template as requested in https://github.com/paritytech/substrate/issues/10892

Fixes #10892 

@kianenigma Please help review to see if this correctly addresses #10892

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

File: bin/node-template/node/Cargo.toml [hunk 1]
```diff
@@ -55,6 +55,9 @@ frame-benchmarking-cli = { version = "4.0.0-dev", path = "../../../utils/frame/b
 # Local Dependencies
 node-template-runtime = { version = "4.0.0-dev", path = "../runtime" }
 
+# CLI-specific dependencies
```
Comment (shawntabrizi) [inline bin/node-template/node/Cargo.toml:58]: There should also be a `try-runtime` feature flag below, and enabling it for the runtime.

Did you test that the binary you build actually does stuff?

**Review by kianenigma — COMMENTED** [review]
Seems fine, should add a feature flag for try-runtime.

File: bin/node-template/node/Cargo.toml (hunk 1 above)
Comment (omadoyeabraham) [reply bin/node-template/node/Cargo.toml:58]: I've made the changes

**Comment by kianenigma** [comment]
Thanks for your contribution @omadoyeabraham! 

Can you make sure this works: 

On your branch, run a `--dev` node, then compile and run `tru-runtime on-runtime-upgrade` and report back the result you see?

**Comment by omadoyeabraham** [comment]
@kianenigma Thanks for your guidance, I had to additionally implement `frame_try_runtime::TryRuntime<Block>` for the node-template Runtime, kindly help verify that the implementation is correct as well.

I've attached an image of the output from the `on-runtime-upgrade` command after making the change.

Screenshot 2022-02-28 at 09 38 46

**Comment by kianenigma** [comment]
now it should be fine, thanks!

**Comment by kianenigma** [comment]
bot merge

**Pull request merged**